### DISCUSSION TITLE
Assets build and deploy successfully to gh-pages branch, but publishing fails

### DISCUSSION THREAD

**Discussion by pltzr** [JamesIves/github-pages-deploy-action#742] [Q&A] [answered]
Hello,

I'm making use of your action on a GitHub Enterprise instance and I'm having a bit of trouble getting my successfully-built VuePress project to "stick".

**To Reproduce**
1. Make a commit to my repository towards the main branch
2. Observe the actions interface that the `build and deploy to gh-pages` job succeeds and that the assets are persisted to the `gh-pages` branch
3. Upon visiting the `Settings` page of my repository, I get the error _Your site is having problems building: Page build failed._ under the "GitHub Pages" section.
4. Upon visiting the GitHub pages endpoint, indeed, it's an out-of-date version.

**Expected behavior**
It should not say **Page build failed :)**

**Workaround**
Curiously, you can work around the problem by doing the following:
1. Change the gh-pages source to "/docs/" and save the changes.
2. Change the gh-pages source back to "/" (root) and save the changes.
3. Now the page will update successfully with the latest version

So of course, with all that I've written above, does anyone know what could be going wrong? Is it possible that the issue lies on my side, on the GitHub Enterprise instance I'm using, or on something else? Very happy if anybody has an idea.

Thank you!

**Comment by pltzr** [comment, marked as answer]
Here's the workflow file by the way:

```bash
# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages

name: Publish Docs

on:
  workflow_dispatch:
  push:
    branches: 
      - master
#    branches:
#    - release/*
  
jobs:
  build:
    runs-on: [ self-hosted ]
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v1
        with:
          node-version: 12

      - name: Install and Build
        run: |
          npm ci
          npm run build
          

  deploy:
    runs-on: [ self-hosted ]
    needs: build
    steps:
      - name: Deploy
        uses: JamesIves/github-pages-deploy-action@4.1.3
        with:
          branch: gh-pages
          folder: .vuepress/dist
```

If it helps, I can also post the output from our build agents - but as mentioned, the build seems to work without issue.

And the `config.js`

```js
module.exports = {
  title: "redacted",
  description: "redacted",
  base: "/redacted/repo-redacted/",

  themeConfig: {
    nav: [
      { text: "Home", link: "/" },
      { text: "About", link: "/about/" }
    ],
    lastUpdated: "Last Updated",
    docsRepo: "https://git.redacted.com/redacted/repo-redacted",
    editLinks: true,
    editLinkText: "Help us improve this page!",
    sidebar: "auto",
    logo: "/favicon.ico"
  },
  head: [["link", { rel: "icon", href: "/favicon.ico" }]],
  plugins: [
    ["@vuepress/active-header-links"],
    [
      "vuepress-plugin-container",
      {
        type: "feature",
        before: (info) => `<div class="feature"><h2>${info}</h2><p>`,
        after: "</p></div>"
      }
    ],
    [
      "container",
      {
        type: "quote"
      }
    ],
    [
      "vuepress-plugin-container",
      {
        type: "feature_box",
        before: '<div class="features">',
        after: "</div>"
      }
    ]
  ]
};
```

**Reply by JamesIves** [reply]
Is the code up-to-date but the live page is not? That error you see in `Settings` indicates that GitHub Pages had an issue re-deploying the page once the branch changed.

This used to be an error on github.com before they made some changes to the default repository token to allow it to trigger page builds, can you instead try adding a `token` input with an access token that is permission scoped to the repository you're deploying to? I'm guessing whatever change GitHub made to tokens didn't make its way to enterprise/on prem instances.

There's some additional information on the token input [here](https://github.com/JamesIves/github-pages-deploy-action#required-setup).

**Reply by pltzr** [reply]
Hi James,

Thanks for reaching out! In the mean time I've got some answers from our local Git Enterprise guys - I think you are right, because here are the errors:

`"GitHub::Pages::Builder::IncompatibleUserTypeError","message":"GitHub Pages builds cannot be executed by Bot accounts."`

and

`"pusher":"github-actions[bot]"`

I'm currently waiting for a technical user to be created and I'll give it another try, also following your token guidelines!

Will update once this is done.

**Reply by pltzr** [reply]
Hey @JamesIves, I added a technical user as a "collaborator" to my repository, and then I used the token field as described - now it works flawlessly. I guess that our Enterprise instance is a little bit behind GH itself!

```yaml
      - name: Deploy
        uses: JamesIves/github-pages-deploy-action@4.1.3
        with:
          branch: gh-pages
          folder: .vuepress/dist
          token: ${{ secrets.TECHNICAL_USER_TOKEN }}
````

Thank you for your help!
